xarray/backends/api.py

```diff
@@ -273,7 +281,8 @@ def maybe_decode_store(store, lock=False):
                                  "default engine or engine='scipy'")
             # if the string ends with .gz, then gunzip and open as netcdf file
             try:
-                store = backends.ScipyDataStore(gzip.open(filename_or_obj))
+                store = backends.ScipyDataStore(filename_or_obj, gzipfile=True,
+                                                autoclose=autoclose)
             except TypeError as e:
```

Comment:
I like moving `gzip.open` into ScipyDataStore. Can you move the try/except clause in there, too?